### PR TITLE
DEV-13404 - apollo must not ignore pod toleration settings on scanners

### PR DESCRIPTION
**Pull request by mayhuberman**

### CODE REVIEW

File: apollo-backend/src/test/java/io/logz/apollo/KubernetesHandlerTest.java [hunk 1]
```diff
@@ -135,16 +149,16 @@ public static void initialize() throws ScriptException, IOException, SQLExceptio
         deploymentDao.updateDeployment(finishedDeploymentForEnvTest.getDeployment());
 
         // Get an instance of the client
-        KubernetesClient kubernetesClient = kubernetesMockClient.replay();
+//        KubernetesClient kubernetesClient = kubernetesMockClient.replay();
```
Comment (roiravhon) [inline apollo-backend/src/test/java/io/logz/apollo/KubernetesHandlerTest.java]: Remove

File: apollo-backend/src/test/java/io/logz/apollo/KubernetesHandlerTest.java [hunk 2]
```diff
@@ -162,7 +176,7 @@ public static void initialize() throws ScriptException, IOException, SQLExceptio
     @Test
     public void testDeploymentMonitor() throws JSONException {
         // We need to wait at least 2 iterations of the monitoring thread + 1 buffer
-        Common.waitABit(5);
+        Common.waitABit(15);
```
Comment (roiravhon) [inline apollo-backend/src/test/java/io/logz/apollo/KubernetesHandlerTest.java]: Why increase here?

File: apollo-backend/src/test/java/io/logz/apollo/KubernetesHandlerTest.java [hunk 3]
```diff
@@ -270,31 +284,37 @@ private static void setMockDeploymentStatus(RealDeploymentGenerator realDeployme
 
         DeploymentStatus deploymentStatus;
         if (finished) {
-            deploymentStatus = new DeploymentStatus(1, 1L, totalReplicas, 0, 1);
+            deploymentStatus = new DeploymentStatus(1, 1, new ArrayList<>(), 1L, 1, totalReplicas, 0, 1);
```
Comment (roiravhon) [inline apollo-backend/src/test/java/io/logz/apollo/KubernetesHandlerTest.java]: The object have changed? does that not affect the deployment status checks?

File: apollo-backend/src/test/java/io/logz/apollo/KubernetesHandlerTest.java (hunk 2 above)
Comment (mayhuberman) [reply apollo-backend/src/test/java/io/logz/apollo/KubernetesHandlerTest.java]: oopsi

File: apollo-backend/src/test/java/io/logz/apollo/KubernetesHandlerTest.java (hunk 3 above)
Comment (mayhuberman) [reply apollo-backend/src/test/java/io/logz/apollo/KubernetesHandlerTest.java]: I changed where it has affected

**Pull request merged**